platform/linux-generic/pktio/dpdk.c

```diff
@@ -1135,6 +1135,13 @@ static void dpdk_init_capability(pktio_entry_t *pktio_entry,
 	rte_eth_dev_info_get(pkt_dpdk->port_id, dev_info);
 	capa->max_input_queues = RTE_MIN(dev_info->max_rx_queues,
 					 PKTIO_MAX_QUEUES);
+
+	/* ixgbe devices support only 16 rx queues in RSS mode */
+	if (!strncmp(dev_info->driver_name, IXGBE_DRV_NAME,
+		     strlen(IXGBE_DRV_NAME)))
+		capa->max_input_queues = RTE_MIN((unsigned)16,
+						 capa->max_input_queues);
+
```

Comment:
Again, shouldn't this sort of configuration be a field in the `dev_info` struct rather than having to be special-cased for each driver? Also, must 16 be a "magic number" here?

Reply:
I'm not sure I got your point but anyway dev_info is type 'struct rte_eth_dev_info' and comes from dpdk, so nothing can be added there.

As @lumag noted this fix is required because of a dpdk defect. With ixgbe drivers rte_eth_dev_info_get() returns a max_rx_queues value which is not valid (at least when using RSS). This is the only driver a have observer this issue with.

The number 16 is explained in the comment line (1139) just above.